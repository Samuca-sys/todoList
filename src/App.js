export default function App() {
  let items = [
    {
      id: 1,
      text: 'Learn JavaScript',
      isCompleted: false,
    },
    {
      id: 2,
      text: 'Learn React',
      isCompleted: false,
    },
    {
      id: 3,
      text: 'Build a React App',
      isCompleted: false,
    }

  ]
  return (
    <div className="container">
      <div className="content">
        <header >
          <h1>THINGS TO DO</h1>
        </header>

        <section>
          <ul className="list">
            {/*render list dinamically using array items  */}
            {items.map(item => (
              <li key={item.id}>
                <label>
                <input type="checkbox" value=""/> {item.text}
                </label>
              </li>
            ))}
          </ul>
        </section>
      </div>
    </div>  
  );
}
