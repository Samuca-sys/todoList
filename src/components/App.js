import React from 'react';

import Header from './Header';
import Footer from './Footer';
import TodoList from './TodoList';
import InputBox from './InputBox';

export default class App extends React.Component {
  constructor() {
    super();
    this.state = { 
      filter: 'active',
      items: [
        {
          id: 1,
          text: 'Learn JavaScript',
          isCompleted: false,
        },
        {
          id: 2,
          text: 'Learn React',
          isCompleted: false,
        },
        {
          id: 3,
          text: 'Build a React App',
          isCompleted: false,
        }
      ]
    }
  }  
  addNew(text) {
    let item = {
      id: this.state.items.length + 1,//next id
      text: text
    };
    let updateList = this.state.items.concat([item]);//join items to item array

    this.setState({
      items: updateList
    })
  }

  handleOnChangeFilter(filter) {
    this.setState({filter})
  }

  render() {
    return (
      <div className="container">
        <div className="content">
          <Header />
          <InputBox addNew={this.addNew.bind(this)}/>
          <TodoList items={this.state.items} filter={this.state.filter}/>
          <Footer 
            count={this.state.items.length} 
            handleOnChangeFilter={this.handleOnChangeFilter.bind(this)}
            {...this.state}
          />
        </div>
      </div>  
    );
  }
}