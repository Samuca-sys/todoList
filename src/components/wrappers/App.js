import Content from "./Content";
import KeyStrokeHandler from "./KeyStrokeHandler";
import StateProvider from "./StateProvider";

export default function App() {
  return (
    <div className="container">
      <div className="content">
        <StateProvider>
          <KeyStrokeHandler>
            <Content />
          </KeyStrokeHandler>
        </StateProvider>
      </div>
    </div>
  );
}
