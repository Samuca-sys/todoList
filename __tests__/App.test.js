import React from "react";
import App from '../src/components/wrappers/App'
import renderer from 'react-test-renderer';

it('renders a snapshot', () => {
  const tree = renderer.create(<App/>).toJSON();
  expect(tree).toMatchSnapshot();
});