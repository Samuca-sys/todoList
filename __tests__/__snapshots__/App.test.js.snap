// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a snapshot 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <header>
      <h1>
        THINGS TO DO
      </h1>
      <input
        autoFocus={true}
        className="add-todo"
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Add New"
        type="text"
        value=""
      />
    </header>
    <section>
      <ul
        className="item-list"
      >
        <li
          className=""
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
             
            Learn JavaScript
          </label>
        </li>
        <li
          className=""
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
             
            Learn React
          </label>
        </li>
        <li
          className=""
        >
          <label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
             
            Build a React App
          </label>
        </li>
      </ul>
    </section>
    <footer>
      <nav>
        <div>
          <img
            alt="add signal"
            className="selected"
            onClick={[Function]}
            src="add.svg"
          />
          <img
            alt="search signal"
            className=""
            onClick={[Function]}
            src="search.svg"
          />
          <strong>
            <span>
              3
               
            </span>
          </strong>
          <span>
            items left
          </span>
        </div>
        <div
          className="filter-list"
        >
          <button
            className="selected"
            onClick={[Function]}
          >
            All
          </button>
          <button
            className=""
            onClick={[Function]}
          >
            Active
          </button>
          <button
            className=""
            onClick={[Function]}
          >
            Completed
          </button>
        </div>
      </nav>
      <p>
        Press 'Esc' to cancel.
      </p>
    </footer>
  </div>
</div>
`;
